core/src/epicli/data/common/ansible/playbooks/roles/kubernetes_master/tasks/cni-plugins/canal.yml

```diff
@@ -35,6 +35,16 @@
         state: reloaded
       when: ansible_facts.services["NetworkManager.service"].state == "running"
 
+- name: Set canal architecture for x86_64
+  set_fact:
+    canal_arch: "amd64"
+  when: ansible_architecture == "x86_64"
+
+- name: Set canal architecture for aarch64
+  set_fact:
+    canal_arch: "arm64"
+  when: ansible_architecture == "aarch64"
+
```

Comment:
How about adding a map into `data\common\ansible\playbooks\group_vars\all.yml`:

```
global_architecture_alias:
  x86_64: amd64
  aarch64: arm64
```

Then we would not need `set_fact` task and the next task could be written like this:
```
- name: Include Canal deployment tasks
  include_tasks: deployments/deploy-template.yml
  vars:
    canal_arch: "{{ global_architecture_alias[ansible_architecture] }}"
    file_name: canal.yml.j2
```

The `global` prefix is used to avoid overriding the global variable by accident.

Reply:
Added.